src/promptflow/tests/sdk_cli_azure_test/e2etests/test_run_operations.py

```diff
@@ -995,3 +995,51 @@ def test_eager_flow_download(self, pf: PFClient, simple_eager_run: Run):
             pf.runs.download(run=run.name, output=tmp_dir)
             for file in expected_files:
                 assert Path(tmp_dir, run.name, file).exists()
+
+    @pytest.mark.skipif(
+        condition=not is_live(),
+        reason="Enable this after fixed sanitizer.",
+    )
+    def test_run_with_compute_instance_session(
+        self, pf: PFClient, compute_instance_name: str, randstr: Callable[[str], str]
+    ):
+        run = Run(
+            flow=Path(f"{FLOWS_DIR}/print_env_var"),
+            data=f"{DATAS_DIR}/env_var_names.jsonl",
+            name=randstr("name"),
+            resources={"compute": compute_instance_name},
+        )
+        rest_run = run._to_rest_object()
+        assert rest_run.compute_name == compute_instance_name
+
+        run = pf.runs.create_or_update(
+            run=run,
+        )
+        assert isinstance(run, Run)
+
+        run = pf.stream(run)
+        assert run.status == RunStatus.COMPLETED
+
+    @pytest.mark.skipif(
+        condition=not is_live(),
+        reason="Enable this after fixed sanitizer.",
+    )
+    def test_run_with_compute_instance_session_yml(
+        self, pf: PFClient, compute_instance_name: str, randstr: Callable[[str], str]
+    ):
+        source = f"{RUNS_DIR}/sample_bulk_run_with_compute_instance.yaml"
+        run_id = randstr("run_id")
+        run = load_run(
+            source=source,
+            params_override=[{"name": run_id}],
+        )
+        rest_run = run._to_rest_object()
+        assert rest_run.compute_name == "my_ci"
+
+        # update ci to actual ci
+        run._resources["compute"] = compute_instance_name
```

Comment:
change this attribute in above `params_override` would also work?

Reply:
Yes, I want to check the loaded YAML file without changing compute name also works, see line 1037.